iothub_client/samples/iothub_convenience_sample/iothub_convenience_sample.c

```diff
@@ -129,13 +173,35 @@ static void send_confirm_callback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void
     (void)printf("Confirmation callback received for message %zu with result %s\r\n", g_message_count_send_confirmations, ENUM_TO_STRING(IOTHUB_CLIENT_CONFIRMATION_RESULT, result));
 }
 
+
+static int check_for_input_func(void *arg)
```

Comment:
@yzhong94 @ewertons 
I'd be interested to know if theres a simpler approach to dropping out of a continuous loop in ANCI C99 without waiting for a potentially lengthy interval(sleep) to complete.